src/software/ai/navigator/obstacle/obstacle_factory.h

```diff
@@ -94,29 +94,44 @@ class ObstacleFactory
      *
      * @return rectangular obstacle
      */
-    ObstaclePtr createObstacleFromRectangle(const Rectangle &rectangle);
+    ObstaclePtr createObstacleFromRectangle(const Rectangle &rectangle) const;
 
    private:
     std::shared_ptr<const ObstacleFactoryConfig> config;
-    double shape_expansion_amount;
+    double obstacle_expansion_amount;
 
     /**
-     * Create circle obstacle directly from a Circle
-     * Note: this helper function does not add a robot radius
+     * Returns an obstacle for the shape expanded on all sides to account for the size of
+     * the robot
      *
-     * @param circle Circle
+     * @param The shape to expand
      *
-     * @return circle shaped obstacle
+     * @return expanded ObstaclePtr
      */
-    ObstaclePtr createObstacle(const Circle &circle);
+    ObstaclePtr expandForRobotSize(const Circle &circle) const;
+    ObstaclePtr expandForRobotSize(const Polygon &polygon) const;
 
     /**
-     * Create polygon obstacle directly from a Polygon
-     * Note: this helper function does not add a robot radius
+     * Returns an obstacle from the Rectangle expanded in the directions in vector
+     * multiplied to account for robot size
      *
-     * @param polygon Polygon
+     * @param rectangle Rectangle to expand
+     * @param directions The list of directions to expand the Rectangle
      *
-     * @return polygon shaped obstacle
+     * @return expanded Rectangle as an obstacle
      */
-    ObstaclePtr createObstacle(const Polygon &polygon);
+    ObstaclePtr expandForRobotSize(const Rectangle &rectangle,
+                                   const std::vector<Vector> &directions) const;
+
+    /**
+     * Returns an obstacle from the Rectangle expanded in the directions in vector
+     * multiplied to account for inflated robot size
+     *
+     * @param rectangle Rectangle to expand
+     * @param directions The list of directions to expand the Rectangle
+     *
+     * @return expanded Rectangle as an obstacle
+     */
+    ObstaclePtr expandForInflatedRobotSize(const Rectangle &rectangle,
```

Comment:
nit: seems somewhat confusing what the difference between `expandForRobotSize` and `expandForInflatedRobotSize`. Obviously one is bigger but it's not super obvious to me when one or the other should be used.

Reply:
How about changing to `expandForDefaultRobotSize` and updating the comments to make it more clear?

Reply:
Hmmm not sure. I guess it's mostly just the context about when using an inflated robot is appropriate, since it's not obvious from the function interface how much bigger an "inflated" robot is. Not sure if "default" would really help that.

@garethellis0 @akhilveeraghanta @EvanMorcom @jsn0716 @LiCody for thoughts.

Reply:
just want to make sure, it seems like its only used in one place for the defense area expansion?

Reply:
yes